src/core/lib/security/credentials/google_default/google_default_credentials.cc

```diff
@@ -89,7 +88,7 @@ grpc_google_default_channel_credentials::create_security_connector(
   bool use_alts =
       is_grpclb_load_balancer || is_backend_from_grpclb_load_balancer;
   /* Return failure if ALTS is selected but not running on GCE. */
-  if (use_alts && !g_is_on_gce) {
+  if (use_alts && !grpc_core::internal::is_on_gce()) {
```

Comment:
@yihuazhang should weigh in on this, but I think this change is undesirable.  The caching was put in here for performance reasons; we do not want to run the potentially expensive caching check every time we create a new google_default_creds instance.  We only need to check whether we're running on GCE once, because after that the result will always be the same.

Reply:
As far as I can tell, this was actually a *double* cache. The concrete implementation provided for `g_gce_tenancy_checker` is [here](https://github.com/grpc/grpc/blob/master/src/core/lib/security/credentials/alts/check_gcp_environment_linux.cc#L59). (the Windows implementation is similarly cached)

Reply:
@markdroth That's correct. `g_is_on_gce` is introduced to avoid unnecessary GCP residency check, which needs to be performed only once. 

@gnossen the performance overhead can also be introduced by lock contention here[https://github.com/grpc/grpc/blob/master/src/core/lib/security/credentials/alts/check_gcp_environment_linux.cc#L58](url) IMO, `g_gce_tenancy_checker` is designed to be called only once.